src/opm/io/eclipse/EclFile.cpp

```diff
@@ -557,57 +527,61 @@ EclFile::EclFile(const std::string& filename) : inputFilename(filename)
 }
 
 
-void EclFile::loadArray(std::fstream& fileH, int arrIndex)
+void EclFile::loadBinaryArray(std::fstream& fileH, int arrIndex)
 {
     fileH.seekg (ifStreamPos[arrIndex], fileH.beg);
 
-    if (formatted) {
-        switch (array_type[arrIndex]) {
-        case INTE:
-            inte_array[arrIndex] = readFormattedInteArray(fileH, array_size[arrIndex]);
-            break;
-        case REAL:
-            real_array[arrIndex] = readFormattedRealArray(fileH, array_size[arrIndex]);
-            break;
-        case DOUB:
-            doub_array[arrIndex] = readFormattedDoubArray(fileH, array_size[arrIndex]);
-            break;
-        case LOGI:
-            logi_array[arrIndex] = readFormattedLogiArray(fileH, array_size[arrIndex]);
-            break;
-        case CHAR:
-            char_array[arrIndex] = readFormattedCharArray(fileH, array_size[arrIndex]);
-            break;
-        case MESS:
-            break;
-        default:
-            OPM_THROW(std::runtime_error, "Asked to read unexpected array type");
-            break;
-        }
+    switch (array_type[arrIndex]) {
+    case INTE:
+        inte_array[arrIndex] = readBinaryInteArray(fileH, array_size[arrIndex]);
+        break;
+    case REAL:
+        real_array[arrIndex] = readBinaryRealArray(fileH, array_size[arrIndex]);
+        break;
+    case DOUB:
+        doub_array[arrIndex] = readBinaryDoubArray(fileH, array_size[arrIndex]);
+        break;
+    case LOGI:
+        logi_array[arrIndex] = readBinaryLogiArray(fileH, array_size[arrIndex]);
+        break;
+    case CHAR:
+        char_array[arrIndex] = readBinaryCharArray(fileH, array_size[arrIndex]);
+        break;
+    case MESS:
+        break;
+    default:
+        OPM_THROW(std::runtime_error, "Asked to read unexpected array type");
+        break;
+    }
 
-    } else {
-        switch (array_type[arrIndex]) {
-        case INTE:
-            inte_array[arrIndex] = readBinaryInteArray(fileH, array_size[arrIndex]);
-            break;
-        case REAL:
-            real_array[arrIndex] = readBinaryRealArray(fileH, array_size[arrIndex]);
-            break;
-        case DOUB:
-            doub_array[arrIndex] = readBinaryDoubArray(fileH, array_size[arrIndex]);
-            break;
-        case LOGI:
-            logi_array[arrIndex] = readBinaryLogiArray(fileH, array_size[arrIndex]);
-            break;
-        case CHAR:
-            char_array[arrIndex] = readBinaryCharArray(fileH, array_size[arrIndex]);
-            break;
-        case MESS:
-            break;
-        default:
-            OPM_THROW(std::runtime_error, "Asked to read unexpected array type");
-            break;
-        }
+    arrayLoaded[arrIndex] = true;
+}
+
+void EclFile::loadFormattedArray(const std::string& fileStr, int arrIndex, long int fromPos)
```

Comment:
There is a quite strong presedence in the opm-code for using `std::size_t` as index and not integers.

Reply:
No problem, will fix